core/src/test/java/org/jboss/pnc/rex/test/GenericHttpClientStandaloneTest.java

```diff
@@ -0,0 +1,198 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rex.test;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.http.Fault;
+import com.github.tomakehurst.wiremock.stubbing.Scenario;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.security.TestSecurity;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+import jakarta.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import org.jboss.pnc.rex.common.enums.Method;
+import org.jboss.pnc.rex.common.exceptions.HttpResponseException;
+import org.jboss.pnc.rex.core.GenericVertxHttpClient;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.net.URI;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@QuarkusTest
+@TestSecurity(authorizationEnabled = false)
+public class GenericHttpClientStandaloneTest {
+
+    private WireMockServer wireMockServer;
+    private static final int MOCK_SERVER_PORT = 8083;
+
+    @BeforeEach
+    void setup() {
+        wireMockServer = new WireMockServer(WireMockConfiguration.options().port(MOCK_SERVER_PORT));
+        wireMockServer.start();
+        WireMock.configureFor("localhost", MOCK_SERVER_PORT);
+    }
+
+    @AfterEach
+    void teardown() {
+        if (wireMockServer != null) {
+            wireMockServer.stop();
+        }
+    }
+
+    @Inject
+    GenericVertxHttpClient httpClient;
+
+    @ParameterizedTest
+    @ValueSource(ints = {425, 429, 500, 503, 599})
+    void shouldRetryOnErrorCodeAndSucceed(int code) throws InterruptedException {
```

Comment:
I guess don't try with 425 cause there's other Test that requires 5s timeout and you shouldn't be able to configure 'max-retries' together with 'timeout'. If that's possible, then it's a bug. 

Optionally, if you want to keep 425 here, you could create a new profile like in `WithoutTaskCleaning.class` and change the config before tests run.